server/src/main/java/org/opensearch/common/compress/CompressorFactory.java

```diff
@@ -61,6 +68,9 @@ public static Compressor compressor(BytesReference bytes) {
             // as a xcontent, we have a problem
             assert XContentHelper.xContentType(bytes) == null;
             return COMPRESSOR;
```

Comment:
I think you should be returning `DEFLATE_COMPRESSOR` here (and checking it above), not `COMPRESSOR`

Reply:
It is ` DEFLATE_COMPRESSOR`. I marked `COMPRESSOR` as deprecated and assigned it to `DEFLATE_COMPRESSOR`. The reason is that `COMPRESSOR`  in code

Reply:
This is why it would make sense to use DEFLATE_COMPRESSOR - COMPRESSOR  should be gone

Reply:
aye aye, sir :-)